### ISSUE TITLE
4 cpu sockets only using 2 sockets..

### ISSUE THREAD

**Issue by TheCyrus2012** [shyba/aeon-stak-cpu#13]
Hi there, Just a quick one. I have a HP DL580 with 4 x E7-4850 CPUs when I set the number of threads the program will only run on 40 of the 80 threads. If I run the program twice it will load in to the same threads.
But if I set the program to have 40 threads and run one and then run the same one again but as admin the 2nd one will load in to the other 40 threads.
Is there anything I have set wrong with this? or is there a problem with the threads/sockets in the code?

Best regards.

**Comment by shyba** [comment]
Can you paste the threads sections of your configuration? I don't know what's wrong as I don't have Windows or multi processor socket machines, but I can take a look at your config to see if it's something easy.

**Comment by TheCyrus2012** [comment]
"cpu_thread_num" : 40,
/
/
/
"cpu_threads_conf" : [ 

	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : false },

],
################################
I have also done it like this 
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 0 },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 1 },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 2 },
	{ "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 3 },

and so on and the problem still happens.

**Comment by shyba** [comment]
Yes, I found a similar issue upstream. See: https://github.com/fireice-uk/xmr-stak-cpu/issues/251

I am working on a rebase so if [xmr-stak-cpu](https://github.com/fireice-uk/xmr-stak-cpu) works for you on cryptonight, then it will also work here after the next release. If not, then we can work together on a fix for the updated code and contribute back there.

**Comment by TheCyrus2012** [comment]
Yeah im good to help test bud thanks. :D

**Comment by TheCyrus2012** [comment]
Do you think you could make one work with DNC2? http://darknetspace.org the pool is http://120.24.15.70/ its only been under a month I tryed youre xmr stak cup it connected but the algorithom is out so it was sending bad info back to the server. I would sooner use xmr stack than the built in miner. Also 1KH on this will get you around 1500 coins a day get in there fast ;)

**Comment by shyba** [comment]
Rebased :)

Here is the latest build for Windows: https://ci.appveyor.com/api/buildjobs/aqaou7xcqjsoeitk/artifacts/aeon-stak-cpu.zip

**Comment by TheCyrus2012** [comment]
Hi Shyba,

I have configered it for DNC2 and it connects but gives back
"result rejected by the pool"
You can use my config to test youre self if you like.
"pool_address" : "120.24.15.70:3333",
"wallet_address" : "D6qqMnUQRTPiVrNgCrn4DYNSbnsX1kbx1FDNYhwEXahS1XXzieLeYY2MXCh1tM4o856epcy7FgSHoZ6VcPhWsCaz8qvaNuD",
"pool_password" : "x",

**Comment by shyba** [comment]
It looks like using wild keccak, not cryptonight-lite.

see: http://darknetspace.org/getstarted.html

**Comment by TheCyrus2012** [comment]
Hi Shyba, Pay no attention to that. They have changed the code. from what I understand DNC floped they have made new DNC2 and that is cryptonight I just got a 1080 mining on it :D 
One thing they have done is let anyone who has DNC coin convert it in to DNC2.
The wallet gui is nice and simple to mine from. One click and you are mining there pool http://120.24.15.70/#
Looks like they have alot of work to do on the website side of things but the wallet is one of the best I have seen. http://darknetspace.org/files/dnc2wallet_2.1.1.zip
I wish I could solo mine it and not have to be in there pool lol.

**Comment by agentpatience** [comment]
is this problem solved?
